docs: badger in config.md (#8662)
ipfs/go-ipfs: docs/config.md

- `flatfs`

  Configures the node to use the flatfs datastore (the default).

  This is the most battle-tested and reliable datastore. 
  You should use this datastore if:

>   - You need a very simple and very reliable datastore and you trust your

Reviewer: ```suggestion
  - You need a very simple and very reliable datastore, and you trust your
```

    filesystem. This datastore stores each block as a separate file in the
    underlying filesystem so it's unlikely to lose data unless there's an issue
    with the underlying file system.
  - You need to run garbage collection in a way that reclaims free space as soon as possible.
  - You want to minimize memory usage.

>   - You are ok with the default speed of data import (or prefer to use `--nocopy`).

Reviewer: ```suggestion
  - You are ok with the default speed of data import, or prefer to use `--nocopy`.
```

  This profile may only be applied when first initializing the node.

- `badgerds`

>   Configures the node to use the experimental badger datastore (warning: uses an outdated badger 1.x).

Reviewer: ```suggestion
  Configures the node to use the experimental badger datastor. Keep in mind that this **uses an outdated badger 1.x**.
```

@lidel is there a specific version number where Badger has become _outdated_? If so, let's add it here in place of `1.x`. If not, just leave the `1.x` in there.

Author: There is badger 3.x now, but we want to remove badger in general. 
This note is only to highlight how old the badger in go-ipfs to discourage people from using it even more.

  Use this datastore if some aspects of performance, 
  especially the speed of adding many gigabytes of files, are critical. However, be aware that:
  
  - This datastore will not properly reclaim space when your datastore is

>     smaller than several gigabytes. If you run IPFS with `--enable-gc` (you have
>     enabled block-level garbage collection), you plan on storing very little data in

Reviewer: ```suggestion
    smaller than several gigabytes. If you run IPFS with `--enable-gc`, you plan on storing very little data in
```

Removed content between the parenthesis - seemed redundant.

    your IPFS node, and disk usage is more critical than performance, consider using
    `flatfs`.

>   - This datastore uses up to several gigabytes of memory.  

Author: I wonder if we should give some examples of repo sizes at which the memory consumption degrades to the point of being useless, so people have a better idea if it even makes sense for their use case.

 Anecdotal evidence: https://github.com/ipfs/distributed-wikipedia-mirror/issues/85
 Do we have more data on this?
